### PR TITLE
Add e2e tests (wdio-vscode-service)

### PR DESCRIPTION
**Pull request by mattseddon**
This PR implements end-to-end tests for the repository. It does not wire the new tests up with the CI (that will be the next PR).

I have used the [wdio-vscode-service](https://webdriver.io/docs/wdio-vscode-service) to do this and used example code from [vscode-marquee](https://github.com/stateful/vscode-marquee/tree/main/test) to make the tests worthwhile.

`yarn test:e2e`:

1. Create the Python venv and install the required packages (runs `yarn setup:venv`)
2. Build the webviews, move them into the appropriate folder and compile the extension (using existing turbo repo commands)
3. Run the end-to-end tests.

The end-to-end test suite performs the following actions:

1. Open the view container and wait for it to load
2. Open the experiments webview
3. Check the webview contains a table
4. Check the experiments webviews table has rows
5. Run an experiment
6. Check the experiments webview has 15 more rows
7. Close the experiments webview
8. Open the plots webview
9. Check the webview has 6 vega plots
10. Check each plot has data

There are no mock/stubs in place so we are relying on the CLI to provide the correct data.

### Demo (video broken into 2 parts because it was too big)

https://user-images.githubusercontent.com/37993418/177745133-c01c43af-a34f-4f77-a4a5-287573a33e08.mov

https://user-images.githubusercontent.com/37993418/177745791-02654ae3-12b6-4344-85a3-5ab02d289e97.mov

### CODE REVIEW

**Review by rogermparent — COMMENTED** [review]
Exciting stuff! Some comments/questions

- It seems this test cannot use the Python extension, so it can't test our integration with it nor can it run in environments that depend on using virtual environments through python (and maybe at all).
- ~~I see this doesn't run on CI yet, does it work? I'd imagine so since the integration tests that spawn a VS Code instance work on CI, but anything could happen.~~ just noticed this was mentioned in the OP

**Review by rogermparent — APPROVED** [review]

File: extension/package.json [hunk 1]
```diff
@@ -1495,6 +1495,7 @@
     "package": "vsce package --yarn -o ./dvc.vsix",
     "test-vscode-build": "tsc -p .",
     "test-vscode-run": "node ./dist/test/runTest.js",
+    "test-e2e": "wdio run ./src/test/e2e/wdio.conf.ts",
```
[the remark this answers is not in the document]
Comment (mattseddon) [reply extension/package.json:1498]: I was really confused by this. The problem is that I was working on this late last night and didn't push my final commit up. What you asked for is [here](https://github.com/iterative/vscode-dvc/pull/1993/files#diff-7ae45ad102eab3b6d7e7896acd08c427a9b25b346470d7bc6507b6481575d519R10). Sorry.

**Comment by mattseddon** [comment]
> > * It seems this test cannot use the Python extension, so it can't test our integration with it nor can it run in environments that depend on using virtual environments through python (and maybe at all).
> 
> I'll start with `dvc.pythonPath` and then try to use the Python extension. I can't remember if it is possible to start the session with another extension installed. I'll try and work it out.

@rogermparent I've looked into the service source code and it is not currently possible to start the service without passing `--disable-extensions`. Look [here](https://github.com/webdriverio-community/wdio-vscode-service/blob/0b56889185110daf59711449387f41da4bbd3493/src/service.ts#L123) and work backwards if you're interested. I've sent the service maintainer/author a DM on the VS Code community slack to ask if he'll accept a PR to change this. Will go with setting the option explicitly for the time being.

**Comment by christian-bromann** [comment]
🎉

**Pull request merged**